baseapp/abci.go

```diff
@@ -360,6 +360,20 @@ func (app *BaseApp) Commit() (res abci.ResponseCommit) {
 	// MultiStore (app.cms) so when Commit() is called is persists those values.
 	app.deliverState.ms.Write()
 	commitID := app.cms.Commit()
+
+	res := abci.ResponseCommit{
+		Data:         commitID.Hash,
+		RetainHeight: retainHeight,
+	}
+
+	// call the hooks with the Commit message
+	for _, streamingListener := range app.abciListeners {
+		if err := streamingListener.ListenCommit(app.deliverState.ctx, res); err != nil {
+			app.logger.Error("Commit listening hook failed", "height", header.Height, "err", err)
+			panic(err)
```

Comment:
Given the above conversation, let's remove this `panic`.  It should be the responsibility of the listener to decide if a panic/halt is necessary on streaming failure.

Reply:
Err, I thought the result of that discussion is listener just decide if propagating the error to consensus state machine, consensus side just treat the error result like other consensus error.

Reply:
I guess your concern is the default behavior should be not panicking, state listener need to explicity panic if want to, that's reasonable too, I'll make that change.